### ISSUE TITLE
Ubuntu18.4，启动貌似成功了，但是没有界面

### ISSUE THREAD

**Issue by myf636** [huan/docker-wechat#28]
![1583151244(1)](https://user-images.githubusercontent.com/13324423/75675651-81d74100-5cc2-11ea-94fd-6e1daebc76b9.png)

**Comment by huan** [comment]
Hmm... interesting...

Please definitely let me know the reason if you can make it work!

**Comment by wangyuhuiever** [comment]
> Hmm... interesting...
> 
> Please definitely let me know the reason if you can make it work!

I rewrite the dochat.sh, use the docker image besewu/wechat, then it work.
test in Ubuntu 19.10

**Comment by yuluobin** [comment]
Same problem here. 
Manjaro & dsiplay scale = 100%
It keeps showing starting but cannot show the window

**Comment by huan** [comment]
@wangyuhuiever It should work under Ubuntu 19.10 because I'm right using this version. If you can reproduce on another Ubuntu 19.10 box, please let me know how to reproduce it.

@yuluobin Linux Manjaro distribution should be another issue. Please feel free to create a new issue for the Manjaro distribution problem. Thank you very much.

**Comment by wangyuhuiever** [comment]
@huan I found some interesting....  I don't know if you have a 21:9 display, it won't work when the external display connection, but will appear when I disconnect the external display.

**Comment by wangyuhuiever** [comment]
I change the display mode, disapper just when I use join Displays, so maybe read the display config error when the container start?

**Comment by kenbing** [comment]
same

**Comment by wangyuhuiever** [comment]
A new discover, I also use your wxwork, but it works fine.

**Comment by wangyuhuiever** [comment]
@huan I found a silly way to fix this bug. 
Refer to [https://blog.csdn.net/weixin_34419326/article/details/94345055?depth_1-utm_source=distribute.pc_relevant.none-task&utm_source=distribute.pc_relevant.none-task](url) this maybe a old bug on GNOME.
Fix this with following step:

1. change the display mode to mirrors ,use `xrandr --output HDMI-1-2 --same-as eDP-1-1`, HDMI-1-2 is the external display,and another is the built in display.

2. dochat.sh

3. change display mode back, use `xrandr --output HDMI-1-2 --right-of eDP-1-1`.
I made a new shell script to do this.

> xrandr --output HDMI-1-2 --same-as eDP-1-1
DOCHAT_SKIP_PULL=true /home/yuhui/App/wechat/dochat.sh &
sleep 5 # during sleep, you must drag the dialog left , otherwise if may stuck on the external display
xrandr --output HDMI-1-2 --right-of eDP-1-1

**Comment by huan** [comment]
@wangyuhuiever Thank you very much for sharing the solution of using DoChat with a dual monitor!

Cloud you please add an FAQ entry in our README by sending a PR, so that the future dual monitor users like you can be easily found the solution?

**Comment by jackielii** [comment]
@wangyuhuiever Indeed! Using one screen it starts up perfectly. in 2 or more monitor setups, it won't start up.

I managed to start it with one monitor, and then switch to multiple monitors. This does work, but after dragging the window to other monitors, it disappears mysteriously

**Comment by ZhangTianrong** [comment]
Same problem here. It says starting but I see no window turning up.

**Comment by Johnny-Wish** [comment]
The above fix doesn't help in my case.
I am trying DoChat out on Ubuntu 18.04 on Parallels virtual machine with an external display connected via HDMI to my physical machine (MacBook pro 15-inch)

I don't think this problem is specific to gnome because I tried on two desktop environments: _gnome_ and _deepin_. After running the script, there is output `Starting Dochat xxxx` and I can see a `DoChat` container running using `sudo docker ps`. However, there is no GUI window that shows up.

**Comment by Johnny-Wish** [comment]
Just tried on a real Ubuntu 18.04 machine. The problem persists. The above fix doesn’t help since I don’t have any external display connected at all.

**Comment by huan** [comment]
Today I run into this issue after I added a new 4K monitor for my Ubuntu 19.10.

And I found that this issue is related to the `zixia/wechat:2.7.1.85`: when I run the v2.7.1.85 in my dual monitor system, the CPU got 100% usage and no window shows up.

However, after I decide to dive into it with my development environment, I found that the v2.8.0.112 works without any problem.

To use 2.8.0.112, you can try the following command by specifying `DOCHAT_WECHAT_VERSION=2.8.0.112`:

```sh
curl -sL https://raw.githubusercontent.com/huan/docker-wechat/master/dochat.sh \
  | DOCHAT_WECHAT_VERSION=2.8.0.112 DOCHAT_DPI=200 bash
```

At last, I found that the input editor in 2.8.0.112 does not show any characters at all, no matter whether it's English or Chinese.

Please feel free reply to this issue if you have any experiences to share, thank you very much.

**Comment by onichandame** [comment]
> Today I run into this issue after I added a new 4K monitor for my Ubuntu 19.10.
> 
> And I found that this issue is related to the `zixia/wechat:2.7.1.85`: when I run the v2.7.1.85 in my dual monitor system, the CPU got 100% usage and no window shows up.
> 
> However, after I decide to dive into it with my development environment, I found that the v2.8.0.112 works without any problem.
> 
> To use 2.8.0.112, you can try the following command by specifying `DOCHAT_WECHAT_VERSION=2.8.0.112`:
> 
> ```shell
> curl -sL https://raw.githubusercontent.com/huan/docker-wechat/master/dochat.sh \
>   | DOCHAT_WECHAT_VERSION=2.8.0.112 DOCHAT_DPI=200 bash
> ```
> 
> At last, I found that the input editor in 2.8.0.112 does not show any characters at all, no matter whether it's English or Chinese.
> 
> Please feel free reply to this issue if you have any experiences to share, thank you very much.

After scanning the QR code it shows that the version of the wechat client is too low and it just don't start.
